src/components/ColorWidget.js

```diff
@@ -0,0 +1,56 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import './ColorWidget.css';
+
+const ColorWidget = ({sender, setColor, type}) => {
+  const [colors, setColors] = useState([
+    { color: 'default', active: true},
+    { color: 'red', active: false },
+    { color: 'orange', active: false },
+    { color: 'yellow', active: false },
+    { color: 'green', active: false },
+    { color: 'blue', active: false },
+    { color: 'purple', active: false },
+  ]);
+
+  const handleColorChange = (e, color) => {
+    setColors(colors.map((current) => {
+      current.active = current.color === color.color;
+      return current;
+    }));
+    setColor(color.color);
+  };
```

Comment:
There's a little duplication of info happening with the active colors. The current active color for a particular user lives in the `localSender` and `remoteSender` state in App.js. But we're also manually keeping in sync which color is active here in this component. If we want a single source of truth, we should set the state in App and in this component we can generate the color list on render and set only the color that matches `sender.color` to active.